MathLib/LinAlg/RowColumnIndices.h

```diff
@@ -14,19 +14,25 @@
 
 namespace MathLib
 {
+// A dummy vector as the default value of the third argument of
+// the constructor.
+static std::vector<std::size_t> dummy_non_ghost_local_ids;
```

Comment:
This (global) variable is not necessary, as far I can understand it. Why not use default initialization of a member variable in the RowColumnIndices ctor?